Server/ConcordiaCurriculumManager/Services/DossierService.cs

```diff
@@ -123,7 +129,42 @@ public async Task DeleteDossier(Guid dossierId)
         _logger.LogInformation($"Deleted {typeof(Dossier)} {dossier.Id}");
     }
 
-    public async Task<Dossier?> GetDossierDetailsById(Guid id) => await _dossierRepository.GetDossierByDossierId(id);
+    public async Task<Dossier?> GetDossierDetailsById(Guid id)
+    {
+        if (!_cacheService.Exists(id.ToString()))
+        {
+            return await GetDossierDetailsAndPersistInCache(id);
+        }
+
+        var dossier = await _dossierRepository.GetDossierByDossierIdWithoutVote(id);
+        if (dossier is null)
+        {
+            return dossier;
+        }
+
+        var votesFromCache = _cacheService.Get(id.ToString());
+        if (votesFromCache is null)
+        {
+            _logger.LogWarning("A cache value has changed or expired while obtaining an entry");
+            return await GetDossierDetailsAndPersistInCache(id);
+        }
+
+        var messages = dossier.Discussion.Messages.ToList();
+        foreach (var vote in votesFromCache)

```

Comment:
This is a clever way of always having the updated dossier contents but still caching the likes.